src/device-registry/utils/create-activity.js

```diff
@@ -314,6 +319,40 @@ const createActivity = {
       const { query } = request;
       const { tenant, deviceName } = query;
 
+      let requestForFilter = {};
+      let previousSiteId = {};
+      requestForFilter["query"] = {};
+      requestForFilter["query"]["device"] = deviceName;
+      let filter = {};
+      const responseFromGenerateFilter = generateFilter.devices(
+        requestForFilter
+      );
+      if (responseFromGenerateFilter.success === false) {
+        return responseFromGenerateFilter;
+      } else {
+        filter = responseFromGenerateFilter.data;
+      }
+
+      const responseFromListDevice = await devicesModel(tenant).list({
+        filter,
+      });
+      if (
+        responseFromListDevice.success === true &&
+        responseFromListDevice.data.length === 1
+      ) {
+        previousSiteId = responseFromListDevice.data[0].site._id;
```

Comment:
During the device recall activity, look up the Site ID and attach it to the device details you are recalling. @BenjaminSsempala @deogratius2phantom